File: Python/09-Student-Coordinator.py
def sortSection(nums): # Function to sort the section in O(N) Time Complexity
    start,end = 0,len(nums)-1
    index = 0
    while start <= end:
        while nums[end] == "M3" and end >= 0:
            end -= 1
        if nums[start] == "M3" and start <= end:
            nums[start], nums[end] = nums[end], nums[start]
            end -= 1
        if nums[start] == "M1":
            nums[start], nums[index] = nums[index], nums[start]
            index += 1
        start += 1

for _ in range(int(input())): # Loop for testcases
    n = int(input()) # Taking input N
    arr = list(map(str,input().split())) # Taking input i.e. Students's Sections
    sortSection(arr) # Calling function to sort the sections
    print(*arr) # Printing the Sorted Sections

File: C/09-Student-Coordinator.c
#include <stdio.h>
#include <string.h>
int main()
{
    int i,n, m1=0, m2=0 ,m3 = 0;
    
    scanf("%d",&n);                     // number of students
    
    for(i=0; i<n; i++ ){
        char str[10];
        scanf("%s",&str);               // input section
        
        if (strcmp(str,"M1")==0)        //  checking for section and incrementing respective counter variable
            m1 = m1 + 1;
        else if (strcmp(str,"M2")==0)
            m2 += 1;
        else if (strcmp(str,"M3")==0)
            m3 += 1;
        
    }
    
    for(int i=0; i<m1; i++)             // printing sections respective number of times.
        printf("M1 ");
    
    for(int i=0; i<m2; i++)
        printf("M2 ");
    
    for(int i=0; i<m3; i++)
        printf("M3 ");
    
    return 0;
}
